### PR TITLE
[feat] Get list of nearby restaurants from servlet

### PR DESCRIPTION
**Pull request by leonyeo**
Servlet that accepts a get request with lat and lng parameters signifying the users current location, and returns a list of restaurants nearby to that location.

https://summer20-sps-48.appspot.com/

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-      ['Din Tai Fung', 'Mcdonalds', 'KFC', 'Subway'];
-
+function getRandomRestaurant(restaurants) {
   // Pick a random restaurant.
   const restaurant = restaurants[Math.floor(Math.random() * restaurants.length)];
 
   // Add it to the page.
   const resElement = document.getElementById('restaurant-name');
-  resElement.innerText = restaurant;
+  resElement.innerHTML = "<p>" + restaurant.name + "</p><p>" + JSON.stringify(restaurant.geometry.location) + "</p>";
```
Comment (lolfuljames) [inline src/main/webapp/script.js:7]: The id tag of the element won't reflect what is actually being written inside.

We can create a new empty HTML element, with the id- "restaurant-loc" and write the location there, or instead, we can change the ID of the p tag in HTML to 'restaurant'.

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -88,6 +85,13 @@ function initMap() {
     curNameElement.innerText = place.name;
     curLocElement.innerText = place.geometry.location.toString();
 
-    getRandomRestaurant();
+    var url = new URL('/places-list', window.location.origin),
+        params = {lat: place.geometry.location.lat(), lng: place.geometry.location.lng()}
+    Object.keys(params).forEach(key => url.searchParams.append(key, params[key]))
```
Comment (lolfuljames) [inline src/main/webapp/script.js:90]: What does this line do?

File: src/main/java/servlets/PlacesListServlet.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        LatLng currentLocation = new LatLng(lat, lng);
+
+        NearbySearchRequest nearbySearch = new NearbySearchRequest(this.context)
+            .location(currentLocation)
+            .radius(500) // Metres, can change via filter
+            .openNow(true)
+            .type(PlaceType.RESTAURANT);
+        
+        try {
+            PlacesSearchResponse nearby = nearbySearch.await(); //Returns top 20 results(One page)
```
Comment (lolfuljames) [inline src/main/java/servlets/PlacesListServlet.java:44]: We need to have check if there is any restaurants returned here. For future testing, can try using NTU Hall 8 as an example.

**Review by lolfuljames — APPROVED** [review]
A few comments, but for the most part it works so I'm approving it.

File: src/main/webapp/script.js (hunk 2 above)
Comment (leonyeo) [reply src/main/webapp/script.js:90]: This is what the fetch standard currently recommends for URL query parameters
https://fetch.spec.whatwg.org/#fetch-api

File: src/main/java/servlets/PlacesListServlet.java (hunk 3 above)
Comment (leonyeo) [reply src/main/java/servlets/PlacesListServlet.java:44]: Thats true I will put in a check for if the list is empty

File: src/main/webapp/script.js (hunk 2 above)
Comment (lolfuljames) [reply src/main/webapp/script.js:90]: Interesting, thanks!

**Pull request merged**